src/TagLib/Ogg/XiphComment.cs

```diff
@@ -409,6 +469,100 @@ protected void Parse (ByteVector data)
 		
 		
 		
+
+#region Private methods
+
+		/// <summary>
+		///    If needed, update the pictures field from the value of the
+		///    pictures array.
+		/// </summary>
+		/// <param name="fieldName">
+		///    Name of the field being queried by the user.
+		///    If the field name is not a picture field name, no update will take place.
+		/// </param>
+		private void EnsurePictureFieldsClean(string fieldName)
+		{
+			if (IsPictureField(fieldName) && picture_fields_dirty)
+				StorePictures();
+		}
+
+		/// <summary>
+		///    Parses the pictures from the COVERART and METADATA_BLOCK_PICTURE
+		///    fields contained in the <see cref="field_list" /> variable.
+		/// </summary>
+		private void ParsePictures()
+		{
+			string[] coverArtStrings = GetField("COVERART"),
+				blockPictureStrings = GetField("METADATA_BLOCK_PICTURE");
+
+			IPicture[] pictures = new IPicture[coverArtStrings.Length + blockPictureStrings.Length];
+
+			// Read old-format COVERART
+			for (int ii = 0; ii < coverArtStrings.Length; ii++)
```

Comment:
Why are you using `ii` instead of just `i`?

Reply:
I just moved the code from the old Pictures getter, this is fixed now.